src/__tests__/SelectMenu.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import SelectMenu from '../SelectMenu'
+import Button from '../Button'
+import {mount} from '../utils/testing'
+import {COMMON} from '../constants'
+import {render as HTMLRender, cleanup} from '@testing-library/react'
+import {axe, toHaveNoViolations} from 'jest-axe'
+import 'babel-polyfill'
+expect.extend(toHaveNoViolations)
+
+const BasicSelectMenu = () => {
+  return (
+    <SelectMenu>
+      <Button as="summary">Projects</Button>
+      <SelectMenu.Modal title="Projects">
+        <SelectMenu.List>
+          <SelectMenu.Item selected href="#">
+            Primer Components bugs
+          </SelectMenu.Item>
+          <SelectMenu.Item href="#">Primer Components roadmap</SelectMenu.Item>
+          <SelectMenu.Divider>stuff</SelectMenu.Divider>
+          <SelectMenu.Item href="#"> Project 3</SelectMenu.Item>
+          <SelectMenu.Item href="#">Project 4</SelectMenu.Item>
+          <SelectMenu.Footer>footer</SelectMenu.Footer>
+        </SelectMenu.List>
+      </SelectMenu.Modal>
+    </SelectMenu>
+  )
+}
+
+const MenuWithTabs = () => {
+  return (
+    <SelectMenu initialTab="Organization">
+      <Button as="summary">Projects</Button>
+      <SelectMenu.Modal title="Projects">
+        <SelectMenu.Tabs>
+          <SelectMenu.Tab index={0} tabName="Repository" />
+          <SelectMenu.Tab index={1} data-test="orgTab" tabName="Organization" />
+        </SelectMenu.Tabs>
+        <SelectMenu.TabPanel tabName="Repository">
+          <SelectMenu.Item href="#">Primer Components bugs</SelectMenu.Item>
+          <SelectMenu.Item href="#">Primer Components roadmap</SelectMenu.Item>
+          <SelectMenu.Item href="#"> Project 3</SelectMenu.Item>
+          <SelectMenu.Item href="#">Project 4</SelectMenu.Item>
+        </SelectMenu.TabPanel>
+        <SelectMenu.TabPanel tabName="Organization">
+          <SelectMenu.Item href="#"> Project 2</SelectMenu.Item>
+        </SelectMenu.TabPanel>
+        <SelectMenu.Footer>Showing 3 of 3</SelectMenu.Footer>
+      </SelectMenu.Modal>
+    </SelectMenu>
+  )
+}
+
+describe('SelectMenu', () => {
+  it('should have no axe violations', async () => {
+    const {container} = HTMLRender(<BasicSelectMenu />)
+    const results = await axe(container)
+    expect(results).toHaveNoViolations()
+    cleanup()
+  })
+  it('implements system props', () => {
+    expect(SelectMenu).toImplementSystemProps(COMMON)
+  })
+
+  it('has default theme', () => {
+    expect(SelectMenu).toSetDefaultTheme()
+  })
+
+  // shallow rendering doesn't work because the component needs `ref` to be defined to render the keyboard hook
+  // using mount returns [Function type] for the type
+  it.skip('respects the "as" prop', () => {
+    expect(mount(<BasicSelectMenu as="span" />).type).toEqual('span')
+  })
+
+  // same as above, returns ReactWraper{} instead of snapshot
+  it.skip('snapshot matches', () => {
+    expect(mount(<BasicSelectMenu />)).toMatchSnapshot()
```

Comment:
`BasicSelectMenu` renders a pretty big DOM tree. Snapshot testing might not be worth it for this one.